### PR TITLE
[AArch64] Implement NEON vamin/vamax intrinsics

### PR DESCRIPTION
**Pull request by momchil-velikov**
This patch implements the intrinsics of the form

    floatNxM_t vamin[q]_fN(floatNxM_t vn, floatNxM_t vm);
    floatNxM_t vamax[q]_fN(floatNxM_t vn, floatNxM_t vm);

as defined in https://github.com/ARM-software/acle/pull/324

### CODE REVIEW

**Review by davemgreen — COMMENTED** [review]
Did you consider emitting `llvm.fmin(llvm.fabs(x), llvm.fabs(y))`?

**Comment by momchil-velikov** [comment]
> Did you consider emitting `llvm.fmin(llvm.fabs(x), llvm.fabs(y))`?

Nope. I'll have a look.

File: llvm/lib/Target/AArch64/AArch64InstrFormats.td [hunk 1]
```diff
@@ -5985,6 +5985,26 @@ multiclass SIMDThreeSameVectorFP<bit U, bit S, bits<3> opc,
         [(set (v2f64 V128:$Rd), (OpNode (v2f64 V128:$Rn), (v2f64 V128:$Rm)))]>;
 }
 
+let mayRaiseFPException = 1, Uses = [FPCR] in
+multiclass SIMDThreeVectorFP<bit U, bit S, bits<3> opc,
```
Comment (CarolineConcatto) [inline llvm/lib/Target/AArch64/AArch64InstrFormats.td:5989]: Why are you adding this class, it is not used by 
defm FAMAX : SIMDThreeSameVectorFP

Comment (momchil-velikov) [reply llvm/lib/Target/AArch64/AArch64InstrFormats.td]: Removed.

**Comment by momchil-velikov** [comment]
Ping?

File: clang/include/clang/Basic/arm_neon.td [hunk 2]
```diff
@@ -2115,3 +2115,8 @@ let ArchGuard = "defined(__aarch64__)", TargetGuard = "lut" in {
     def VLUTI4_BF_X2_Q   : SInst<"vluti4_laneq_x2", ".2(<U)I", "Qb">;
   }
 }
+
+let ArchGuard = "defined(__aarch64__)", TargetGuard = "faminmax" in {
```
Comment (SpencerAbson) [inline clang/include/clang/Basic/arm_neon.td:2119]: Should this target guard not be `"neon,faminmax"`? We are currently testing the use of these intrinsics under `+neon,+faminmax` and `FeatureFAMINMAX` does not include `FeatureNEON` as a dependency.

Comment (momchil-velikov) [reply clang/include/clang/Basic/arm_neon.td]: Done

**Review by CarolineConcatto — APPROVED** [review]

**Pull request merged**